diesel_derives/src/queryable_by_name.rs

```diff
@@ -43,15 +43,30 @@ pub fn derive(item: syn::DeriveInput) -> Tokens {
 }
 
 fn build_expr_for_model(model: &Model) -> Tokens {
-    let table_name = model.table_name();
     let attr_exprs = model.attrs.iter().map(|attr| {
         let name = attr.field_name();
         let column_name = attr.column_name();
-        let st = quote!(diesel::dsl::SqlTypeOf<#table_name::#column_name>);
+        let st = sql_type(&attr, model);
         quote!(#name: diesel::row::NamedRow::get::<#st, _>(row, stringify!(#column_name))?)
     });
 
     quote!(Self {
         #(#attr_exprs,)*
     })
 }
+
+fn sql_type(attr: &Attr, model: &Model) -> Tokens {
+    let table_name = model.table_name();
+    let column_name = attr.column_name();
+
+    match attr.sql_type() {
+        Some(st) => quote!(#st),
+        None => if model.has_table_name_annotation() {
+            quote!(diesel::dsl::SqlTypeOf<#table_name::#column_name>)
+        } else {
+            panic!(
```

Comment:
Instead of panicing in error cases we should emit a `compile_error!` in place of the actual `impl QueryableByName`. This should result in better readable error messages.

Reply:
The error message is the same either way. Panicking is how proc macros communicate errors to the user.

Reply:
<img width="1081" alt="screen shot 2017-11-30 at 10 55 13 am" src="https://user-images.githubusercontent.com/1529387/33446348-f7a4cbee-d5bc-11e7-8f6b-80fcc936c288.png">

Reply:
I guess having the message at the top is marginally better